jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
@pierre 
You meant to pass both paymentExternalKey and transactionExternalKey parameters as null?
There is a check [here](https://github.com/javier-gomez-hs/killbill/blob/e7c1e41dc2492843cbf5af7c57972f3b6fc053ef/payment/src/main/java/org/killbill/billing/payment/api/DefaultPaymentApi.java#L295-L295) to verify that the transactionExternalKey is not null.
Should I remove that or send as null the paymentExternalKey and a random UUID as the transactionExternalKey?